src/selection.coffee

```diff
@@ -183,7 +183,7 @@ class Selection extends Model
 
   # Public: Clears the selection, moving the marker to the head.
   clear: ->
-    @marker.setProperties(goalBufferRange: null)
+    @marker.setProperties(goalBufferRange: null, goalScreenRange: null)
```

Comment:
What's your thinking behind retaining the `goalBufferRange` here now that you've added `goalScreenRange`? Are we still using it in other ways?